### PR TITLE
React 18 support

### PR DESCRIPTION
**Pull request by atomiks**
Closes https://github.com/floating-ui/react-popper/issues/349

### CODE REVIEW

**Comment by layershifter** [comment]
@atomiks friendly ping, any estimates when this could be released?

**Review by FezVrasta — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -93,16 +94,16 @@
...
     "jest": "^25.2.4",
     "parcel-bundler": "^1.12.4",
     "prettier": "^2.0.2",
     "pretty-quick": "^2.0.1",
-    "react": "17.0.0",
-    "react-dom": "^17.0.0",
+    "react": "18.0.0",
+    "react-dom": "^18.0.0",
     "react-spring": "^8.0.27",
     "react-test-renderer": "^17.0.0",
```
Comment (chentsulin) [inline package.json:108]: Should react-test-renderer be updated to 18 too?

**Comment by aliveli186** [comment]
When will React 18 be supported?
Upgraded and stuck.

**Comment by FezVrasta** [comment]
I need to find some time to make this PR pass the tests, if anyone wants to contribute please do so 🙇  otherwise I'll try to do it as soon I have some time available

**Comment by chentsulin** [comment]
Hi @FezVrasta, I created #441 and made some changes to fix the tests.

**Comment by chentsulin** [comment]
@FezVrasta How about releasing a new minor version based on this PR?

**Comment by FezVrasta** [comment]
Waiting to understand if we also need https://github.com/floating-ui/react-popper/pull/439

**Comment by aliveli186** [comment]
> Waiting to understand if we also need #439

Ready to release? We're stuck.

**Comment by atomiks** [comment]
#439 is not relevant for now, just release @FezVrasta

**Comment by FezVrasta** [comment]
2.3.0 is now live with these changes

**Comment by varand-pez** [comment]
I'm using Semantic UI React and found a thread about this issue in that repo which led me here. Just wanted to say this is still happening. I'm using 
`semantic-ui 2.13` which uses `react-poper 2.3.0` and `react 18.2.0`

**Pull request merged**